File: api/Models/Flight.cs
namespace Project_1_Web_API.Models
{
    public class Flight
    {
        public int Id { get; set; }
        public DateTime DepartureTime { get; set; }
        public string DepartureAirport { get; set; }
        public DateTime ArrivalTime { get; set; }
        public string ArrivalAirport { get; set; }
        public int MaxCapacity { get; set; }

        // Relational properties
        public virtual ICollection<Booking>? Bookings { get; set; }

        // Not stored data
        public int PassengersBooked => Bookings?.Count ?? 0;
    }
}

File: client/src/app/models/flight.ts
export interface Flight {
    id: number;
    departureTime: any;
    departureAirport: string;
    arrivalTime: any;
    arrivalAirport: string;
    maxCapacity: number;
    passengersBooked:number;
}

export interface FlightDTO {
    departureTime: any;
    departureAirport: string;
    arrivalTime: any;
    arrivalAirport: string;
    maxCapacity: number;
}